### PR TITLE
implement parameter use_auto_model_for_sequence_classification 

### PR DESCRIPTION
**Pull request by ArneBinder**
... for `SequenceClassificationModel`. If enabled, `AutoModelForSequenceClassification` will be used instead of `AutoModel` + costume classification head (and pooler). 

Especially, this allows to fine-tune Llama models. Example call (for `pie-document-level`):
```
python src/train.py \
experiment=sciarg_relations \
taskmodule.tokenizer_name_or_path=TheBloke/Llama-2-7B-GPTQ \
taskmodule.max_window=2048 \
model.model_name_or_path=TheBloke/Llama-2-7B-GPTQ \
+model.use_auto_model_for_sequence_classification=true \
+model.freeze_base_model=true \
+model.model_kwargs.device_map=auto \
+model.model_kwargs.torch_dtype=auto \
trainer=gpu \
datamodule.batch_size=1
```
Notes: 
- This requires the recently implemented QPTQ integration for Huggingface `transformers`, see [here](https://huggingface.co/blog/gptq-integration#native-support-of-gptq-models-in-%F0%9F%A4%97-transformers), i.e.
  - [transformers>=4.32.0](https://github.com/huggingface/transformers/releases/tag/v4.32.0): `pip install transformers==4.32.1` 
  - [AutoQPTQ](https://github.com/PanQiWei/AutoGPTQ): `pip install auto-gptq`
  - Optimum: `pip install optimum`
  - requires to run on GPU!
-  `model.freeze_base_model=true` will **not** freeze the embedding layer (we resize the embedding layer after freezing the base model which unfreezes the embedding layer again)

EDIT: Not sure, if that really learns sth... the loss is nan, at least for the first >6k steps.

### CODE REVIEW

**Comment by ArneBinder** [comment]
This is is too complicated. We use #16 in combination with #20 instead. 

Note, that the hints regarding fine-tuning with Llama 2 from the PR description above should also be applicable to that.
